framework/src/utils/InputParameters.C

```diff
@@ -500,7 +500,7 @@ InputParameters::defaultCoupledValue(const std::string & coupling_name, unsigned
                "variable added with params.addRequiredCoupledVar() \n 3. The call to get the "
                "coupled value should have been properly guarded with isCoupled()\n");
 
-  return value_it->second._coupled_default[i];
+  return value_it->second._coupled_default.at(i);
```

Comment:
Do we need this? I'm not a big fan of `at()` because it throws and in MPI land it doesn't call MPI_Abort which can complicate tracking failures down. We have the check above so do we need this?

Reply:
The above check only looks at whether there are *any* coupled values for the `coupling_name`; it doesn't look to see whether the component `i` is within the `_coupled_default` bounds. I could add another method like `hasDefaultCoupledComponent` that I check in the relevant `Coupleable` methods if you like that better